components/authz-service/storage/postgres/migration/sql/47_staged_rules_tables.up.sql

```diff
@@ -0,0 +1,57 @@
+BEGIN;
+
+CREATE TABLE iam_staged_project_rules (
+  db_id SERIAL PRIMARY KEY,
+  id TEXT NOT NULL UNIQUE,
+  project_id TEXT REFERENCES iam_projects ON DELETE CASCADE,
+  name TEXT NOT NULL,
+  type TEXT NOT NULL,
+  deleted BOOLEAN NOT NULL DEFAULT FALSE
+);
+
+CREATE TABLE iam_staged_rule_conditions (
+  db_id SERIAL PRIMARY KEY,
+  rule_db_id INTEGER REFERENCES iam_staged_project_rules ON DELETE CASCADE,
+  value TEXT[] NOT NULL,
+  attribute TEXT NOT NULL,
+  operator TEXT NOT NULL,
+  deleted BOOLEAN NOT NULL DEFAULT FALSE
+);
+
+CREATE OR REPLACE FUNCTION
+  query_staged_rule(_rule_db_id TEXT, _project_filter TEXT[])
+  RETURNS json AS $$
+
+  WITH t AS (
+    SELECT
+      r.id,
+      r.project_id,
+      r.name,
+      r.type,
+      r.deleted,
+      -- A rule can't exist without conditions so we don't need to worry
+      -- about null case here.
+      json_agg(rc) AS conditions
+    FROM iam_staged_project_rules AS r
+    LEFT OUTER JOIN iam_staged_rule_conditions
+    AS rc ON rc.rule_db_id=r.db_id
+    WHERE id=_rule_db_id AND projects_match_for_rule(project_id, _project_filter)
+    GROUP BY r.id, r.project_id, r.name, r.type, r.deleted
+  )
+  SELECT row_to_json(t) AS rule FROM t;
+
+$$ LANGUAGE sql;
+
+CREATE OR REPLACE FUNCTION
+  query_rule_table_associations(_id TEXT)
```

Comment:
behavior changes contingent on whether the rule exists in current/staging/both, hence the need for this helper function